dot/network/notifications.go

```diff
@@ -289,16 +289,15 @@ func (s *Service) sendData(peer peer.ID, hs Handshake, info *notificationsProtoc
 			return
 		}
 
-		// TODO: ensure grandpa stores *all* previously received votes and discards them
-		// only when they are for already finalised rounds; currently this causes issues
-		// because a vote might be received slightly too early, causing a round mismatch err,
-		// causing grandpa to discard the vote. (#1855)
-		_, isConsensusMsg := msg.(*ConsensusMessage)
-		if !added && !isConsensusMsg {
+		// if we could not add message in cache, message was already present in the
+		// cache and was sent.
+		if !added {
```

Comment:
Should we rename `added` to `alreadyInCache`?

Reply:
Actually, that is not the meaning of this variable. 

In `added, err := s.host.messageCache.put(peer, msg)`, if message is already in cache, put method does not add the message and `added` is false.  If we could add, added will be true.

So, it sort of like `!added` is similar to `alreadyInCache`. But, value of `added` does not depend just on whether the message was in cache or not. If `put` fails for some other reason, that would also impact the value of `added`.

so, `alreadyInCache := !added & err !=nil`